### PR TITLE
Fix cmake scripts to work with new VS dogfood

### PR DESCRIPTION
**Pull request by jkotas**
The default layout have changed and so the hardcoding *.dir for the location of .pdb for static libraries does not work anymore

### CODE REVIEW

File: src/coreclr/nativeaot/Directory.Build.props [hunk 1]
```diff
@@ -45,6 +45,9 @@
     <DisableImplicitConfigurationDefines>true</DisableImplicitConfigurationDefines>
 
     <DocumentationFile>$(OutputPath)$(MSBuildProjectName).xml</DocumentationFile>
+
+    <!-- Send .deps.json to intermediate directory to avoid polluting published package -->
```
Comment (jkotas) [inline src/coreclr/nativeaot/Directory.Build.props:49]: Unrelated fix that I made along the way

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!
